plan/stats.go

```diff
@@ -110,11 +109,10 @@ func (p *baseLogicalPlan) deriveStats() (*statsInfo, error) {
 	return profile, nil
 }
 
-func (ds *DataSource) getStatsByFilter(conds expression.CNFExprs) *statsInfo {
-	profile := &statsInfo{
+func (ds *DataSource) getStatsByFilter(conds expression.CNFExprs) (profile *statsInfo, histColl *statistics.HistColl) {
```

Comment:
It seems that all the code changes related to the functions of `DataSource` is not irrelevant?

Reply:
Yes, we can extract the same part with `selection.deriveStats`